packages/react/src/internal/components/UnderlineTabbedInterface.tsx

```diff
@@ -0,0 +1,239 @@
+// Used for UnderlineNav and UnderlinePanels components
+
+import React, {forwardRef, type FC, type PropsWithChildren} from 'react'
+import type {IconProps} from '@primer/octicons-react'
+import styled, {keyframes} from 'styled-components'
+import CounterLabel from '../../CounterLabel'
+import sx, {type SxProp} from '../../sx'
+import type {ForwardRefComponent as PolymorphicForwardRefComponent} from '../../utils/polymorphic'
+import {defaultSxProp} from '../../utils/defaultSxProp'
+
+// The gap between the list items. It is a constant because the gap is used to calculate the possible number of items that can fit in the container.
+export const GAP = 8
+
+export const StyledUnderlineWrapper = styled.div`
+  display: flex;
+  padding-inline: var(--stack-padding-normal);
+  justify-content: flex-start;
+  align-items: center;
+  /* make space for the underline */
+  min-height: var(--control-xlarge-size);
+  /* using a box-shadow instead of a border to accomodate 'overflow-y: hidden' on UnderlinePanels */
+  box-shadow: inset 0px -1px var(--borderColor-muted);
+
+  ${sx};
+`
+
+export const StyledUnderlineTabList = styled.ul`
```

Comment:
Similar naming suggestion: to make it more generic, how would you feel about renaming this to `StyledUnderlineList`?